### PR TITLE
the queue

### PR DESCRIPTION
**Pull request by alexjhawk**

### CODE REVIEW

**Comment by alexjhawk** [comment]
Hold off on review until the other PR (google code style) is approved and merged, so this PR can have that test run.

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]:
```suggestion
  private static long queueFifoTimeSpanMins = 1;
```

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]:
```suggestion
  private static synchronized long getQueueFifoTimeSpanMillis() {
```

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]:
```suggestion
  public static synchronized long getQueueFifoTimeSpanMins() {
```

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]:
```suggestion
  public static synchronized void setQueueFifoTimeSpanMins(long timeSpanMins) {
```

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]:
```suggestion
    queueFifoTimeSpanMins = timeSpanMins;
```

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]:
```suggestion
    if ((startTimePtrLong + getQueueFifoTimeSpanMillis()) > getCurrentTime()) {
```

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]:
```suggestion
      endTimePtr = "" + (startTimePtrLong + getQueueFifoTimeSpanMillis());
```

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: changed

**Review by oliver-walker-hms — DISMISSED** [review]

**Review by TomKimsey — CHANGES_REQUESTED** [review]
I dont like the use of the word pointer or (ptr) for time tracking since it could be confused for a c style pointer. Especially since this code will be used by people outside of hms who may not know java well.

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: changed

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: changed

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: changed

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: changed

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: changed

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: changed

**Comment by alexjhawk** [comment]
> I dont like the use of the word pointer or (ptr) for time tracking since it could be confused for a c style pointer. Especially since this code will be used by people outside of hms who may not know java well.

@TomKimsey What do you think is a better name? I couldn't think of anything else, which is how I ended up with pointer/ptr

**Comment by oliver-walker-hms** [comment]
I was using time tracker for the same reason. timeTracker or queueTimeTracker would work

**Comment by TomKimsey** [comment]
> l

This came up in Nordson as well and I forget what Oliver changed it to.

Marker, index, tracker, position, value

**Comment by alexjhawk** [comment]
> > l
> 
> This came up in Nordson as well and I forget what Oliver changed it to.
> 
> Marker, index, tracker, position, value

changing to tracker

**Comment by TomKimsey** [comment]
Pointer is descriptive but i think some of the people who use this might not know java well and get confused.

**Comment by alexjhawk** [comment]
changed pointer to tracker

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: Comment needs revising "Ptr"

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]:
```suggestion
    if ((startTimeTrackerLong + getQueueFifoTimeSpanMillis()) > getCurrentTime()) {
```

Comment (TomKimsey) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: again

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: Might be more clear to break this out into two lines for clarity.

Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]:
```suggestion
      long lastTimeReadMillis = startTimeTrackerLong + getQueueFifoTimeSpanMillis();
      endTimeTracker = Long.toString(lastTimeReadMillis);
```

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: units?

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: units?

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: Long.toString() should be used here as im pretty sure its getting called anyway. Then you dont have to have the flexy do the string concat

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: Long.toString() 

See other comment

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]:
```suggestion
   * @return current time as long in milliseconds
```

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: revise comment to not use pointer

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: pointer

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: pointer, revise comment

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: Why does this exist?

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: units

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: units

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]:
```suggestion 
    final String newTimeTrackerVal = Long.toString(endTimeTrackerLong + 1);
```

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: should these be moved to their own class or even library? They are pretty generic.

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: i think this method should get refactored so that the native long is maintained in scope to avoid this unnecessary conversion.

**Review by TomKimsey — CHANGES_REQUESTED** [review]

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (TomKimsey) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: OR
```suggestion
      endTimeTracker = Long.toString(startTimeTrackerLong + getQueueFifoTimeSpanMillis());
```

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: sorry, missed the parentheses first time. Only noticed the method capitalization change. Will fix

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: done

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: moved to variable. Parentheses no longer applicable. Fixed

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: done

**Review by oliver-walker-hms — DISMISSED** [review]

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: done

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: done

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: done

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: done

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: done

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: done

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: done

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: I made this separate because it is used in multiple locations, and in the previous implementation that Oliver created, it was a separate method that added an offset of time. This allows a quick modification to have the same offset functiionality

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: Units isn't really applicable here. ebdStartTime is a formatted date string: ddMMyyyy_HHmmss

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: Units isn't really applicable here. ebdEndTime is a formatted date string: ddMMyyyy_HHmmss

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: done

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (oliver-walker-hms) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: at one point in the nordson project this was used to subtract a few seconds from the system time when getting the current time. That was to see if 1300+ tags was creating a lag in writing to the historical logs. I think this also had the timezone offset logic in here in the nordson project

Comment (TomKimsey) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: But that offset is not currently being used? If its not being used it probably shouldnt go in the repo.

Comment (oliver-walker-hms) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: than aren't we stuck with UTC? I feel like the timezone code should probably get added in eventually

Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: removed (for now at least)

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: I can move these to a library. Do you think these belong in any of the existing libraries, or should I do a new one?

Comment (TomKimsey) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: I dont think they quite fit in the string lib. Maybe sc-flexy-file-tools

Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: moved

File: src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java
Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/historicaldata/HistoricalDataQueueManager.java]: refactored

**Review by TomKimsey — APPROVED** [review]

**Review by oliver-walker-hms — APPROVED** [review]

**Comment by TomKimsey** [comment]
Moving the string file stuff to another file really cleaned this up.

**Pull request merged**
